presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -71,7 +49,36 @@
 
     public JdbcRecordCursor(JdbcClient jdbcClient, JdbcSplit split, List<JdbcColumnHandle> columnHandles)
     {
-        this.columnHandles = ImmutableList.copyOf(requireNonNull(columnHandles, "columnHandles is null"));
+        this.columnHandles = requireNonNull(columnHandles, "columnHandles is null").stream()
+                .toArray(JdbcColumnHandle[]::new);
+
+        booleanReadFunctions = new BooleanReadFunction[this.columnHandles.length];
+        doubleReadFunctions = new DoubleReadFunction[this.columnHandles.length];
+        longReadFunctions = new LongReadFunction[this.columnHandles.length];
+        sliceReadFunctions = new SliceReadFunction[this.columnHandles.length];
+
+        for (int i = 0; i < this.columnHandles.length; i++) {
+            ReadMapping readMapping = jdbcClient.toPrestoType(this.columnHandles[i].getJdbcTypeHandle())
+                    .orElseThrow(() -> new IllegalStateException("Unsupported column type"));
```

Comment:
Let's make these `VerifyException` since this is not *"a method has been invoked at an illegal or inappropriate time"*